ballista/scheduler/src/state/task_manager.rs

```diff
@@ -75,41 +75,31 @@ type ActiveJobCache = Arc<DashMap<String, JobInfoCache>>;
 
 #[derive(Default)]
 struct ActiveJobQueue {
-    queue: SegQueue<String>,
+    queue: Mutex<BinaryHeap<ActiveJob>>,
     jobs: ActiveJobCache,
 }
 
 impl ActiveJobQueue {
-    pub fn pop(&self) -> Option<ActiveJobRef> {
-        loop {
-            if let Some(job_id) = self.queue.pop() {
-                if let Some(job_info) = self.jobs.get(&job_id) {
-                    return Some(ActiveJobRef {
-                        queue: &self.queue,
-                        job: job_info.clone(),
-                        job_id,
-                    });
-                } else {
-                    continue;
-                }
-            } else {
-                return None;
-            }
-        }
+    pub fn pop(&self) -> Option<ActiveJob> {
+        self.queue.lock().pop()
     }
 
     pub fn pending_tasks(&self) -> usize {
         let mut count = 0;
         for job in self.jobs.iter() {
-            count += job.pending_tasks.load(Ordering::Acquire);
+            count += job.available_tasks.load(Ordering::Acquire);
```

Comment:
is it correct?

Reply:
I renamed it to be consistent with the execution graph, where this number is coming from. If we change the implementation we might as well rename this again.